Sources/Fuzzilli/Mutators/CodeGenMutator.swift

```diff
@@ -31,6 +31,10 @@ public class CodeGenMutator: BaseInstructionMutator {
     
     public override func mutate(_ instr: Instruction, _ b: ProgramBuilder) {
         b.adopt(instr, keepTypes: true)
+        guard b.context.contains(.script) else {
```

Comment:
I think this should go into `canMutate` and just be an `assert` here

Reply:
Currently `canMutate` doesn't have a reference to the active `ProgramBuilder` instance. We can pass this as a parameter along with the instruction but that would mean changing `canMutate` in the parent class (i.e. `BaseInstructionMutator`). Should I go ahead and do that?

Reply:
Ah right, yeah that won't work because the `ProgramBuilder` will be empty and so won't be able to provide useful context information :( Maybe for the splicing algorithm it'll turn out to be useful/necessary to have context information associated with every instruction in a `Program` (i.e. a `Array<Context>` in every `Program`), and then we'd also have it available here. So let's revisit this later